modules/webp/SCsub

```diff
@@ -142,6 +142,20 @@ if env["builtin_libwebp"]:
 
     env_thirdparty = env_webp.Clone()
     env_thirdparty.disable_warnings()
+
+    # libwebp assumes that any MSVC compatible compiler has SSE4.1 intrinsics always enabled.
+    # This causes problems while building with Clang, we need to add a compiler flag for few source files.
+    if env["arch"] in ("x86_64", "x86_32") and env.msvc and env["use_llvm"]:
+        sse41_sources = []
+        for file in thirdparty_sources[:]:
+            if file.endswith("_sse41.c"):
+                sse41_sources.append(file)
+                thirdparty_sources.remove(file)
+
+        env_sse41 = env_thirdparty.Clone()
+        env_sse41.Append(CCFLAGS=["-msse4.1"])
+        env_sse41.add_source_files(thirdparty_obj, sse41_sources)
```

Comment:
I would prefer to patch libwebp to properly detect things like GCC/Clang do with MinGW or on Unix.

We don't enable SSE4.1 intrinsics currently for our GCC/Clang builds, and so doing this here would potentially reduce the supported hardware for Godot binaries made with clang-cl.

We can consider enabling SSE4.1 for all platforms and drop support for older CPUs, but this should be evaluated separately from this PR adding support for a new compiler.

I would assume something like this should fix it:
```diff
diff --git a/thirdparty/libwebp/src/dsp/cpu.h b/thirdparty/libwebp/src/dsp/cpu.h
index c86540f280..4dbe607aec 100644
--- a/thirdparty/libwebp/src/dsp/cpu.h
+++ b/thirdparty/libwebp/src/dsp/cpu.h
@@ -47,12 +47,12 @@
 // x86 defines.
 
 #if !defined(HAVE_CONFIG_H)
-#if defined(_MSC_VER) && _MSC_VER > 1310 && \
+#if defined(_MSC_VER) && !defined(__clang__) && _MSC_VER > 1310 && \
     (defined(_M_X64) || defined(_M_IX86))
 #define WEBP_MSC_SSE2  // Visual C++ SSE2 targets
 #endif
 
-#if defined(_MSC_VER) && _MSC_VER >= 1500 && \
+#if defined(_MSC_VER) && !defined(__clang__) && _MSC_VER >= 1500 && \
     (defined(_M_X64) || defined(_M_IX86))
 #define WEBP_MSC_SSE41  // Visual C++ SSE4.1 targets
 #endif
```
and could be PR'ed upstream too.